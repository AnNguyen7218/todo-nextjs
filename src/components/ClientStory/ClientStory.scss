@import 'config.colors';
@import 'config.gutters';
@import 'config.fonts';
@import 'config.media';

$avatar-width-md: 15rem;
$avatar-width-xl: 24.4rem;

$padding-left-client-quote-sm: gutter(16);
$padding-left-client-quote-md: gutter(20);
$padding-left-client-quote-xl: gutter(32);

.container {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  &--light {
    background-color: $light-gray;
    color: $black;
  }

  &--dark {
    background-color: $black;
    color: $white;
  }

  @include break(md) {
    flex-direction: row;
  }
}

.avatar {
  width: 100%;
  object-fit: contain;

  @include break(md) {
    object-fit: cover;
    width: $avatar-width-md;
    height: 21.6rem;
    object-position: center;
  }

  @include break(xl) {
    width: $avatar-width-xl;
    height: 24rem;
  }
}

.client-quote {
  font-weight: 500;
  padding: $padding-left-client-quote-sm gutter(16) gutter(84) gutter(16);
  min-height: 10rem;

  @include break(md) {
    padding-left: $padding-left-client-quote-md;
  }

  @include break(xl) {
    padding-left: $padding-left-client-quote-xl;
    padding-top: gutter(24);
    padding-right: gutter(32);
  }
}

.client {
  position: absolute;
  bottom: gutter(8);
  left: $padding-left-client-quote-sm;
  right: gutter(16);
  min-width: 22rem;
  @include small-medium;

  @include break(md) {
    left: calc(#{$padding-left-client-quote-md} + #{$avatar-width-md});
  }

  @include break(xl) {
    left: calc(#{$padding-left-client-quote-xl} + #{$avatar-width-xl});
    bottom: gutter(24);
    right: gutter(32);
  }
}

.client-title {
  @include small;
}
