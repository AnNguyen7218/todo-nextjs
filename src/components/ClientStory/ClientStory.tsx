import React from 'react';
import cx from 'classnames';
import css from './ClientStory.scss';

type Props = {
  client: ClientStoryType;
  theme?: 'light' | 'dark';
};

const ClientStory: React.FC<Props> = ({
  client: { avatar, clientName, clientQuote, clientTitle, company },
  theme = 'light',
}) => {
  return (
    <div className={cx(css['container'], css[`container--${theme}`])}>
      <img src={avatar} alt="client-avatar" className={css['avatar']} />
      <p className={css['client-quote']}>&quot;{clientQuote}&quot;</p>
      <p className={css['client']}>
        {clientName},
        <br />
        <span className={css['client-title']}>
          {clientTitle} {company}
        </span>
      </p>
    </div>
  );
};

export default ClientStory;
