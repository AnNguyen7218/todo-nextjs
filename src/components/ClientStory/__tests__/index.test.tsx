import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ClientStory from '../ClientStory';
import Client2 from 'assets/images/client2.jpg';

afterEach(cleanup);

const client = {
  avatar: Client2,
  clientName: 'Christina',
  clientTitle: 'Product Designer',
  company: 'company',
  clientQuote:
    'Amet rhoncus commodo san, sagittis enim. Porttitor in cursus eu amet sit.',
};

describe('Slider Component', () => {
  test('Slider should match snapshot', () => {
    const { container } = render(<ClientStory client={client} />);
    expect(container).toMatchSnapshot();
  });
});
