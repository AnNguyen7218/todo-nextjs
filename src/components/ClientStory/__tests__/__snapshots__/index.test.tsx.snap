// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component Slider should match snapshot 1`] = `
<div>
  <div
    class="container container--light"
  >
    <img
      alt="client-avatar"
      class="avatar"
      src="client2.jpg"
    />
    <p
      class="client-quote"
    >
      "
      Amet rhoncus commodo san, sagittis enim. Porttitor in cursus eu amet sit.
      "
    </p>
    <p
      class="client"
    >
      Christina
      ,
      <br />
      <span
        class="client-title"
      >
        Product Designer
         
        company
      </span>
    </p>
  </div>
</div>
`;
